packages/store-sync/src/recs/recsStorage.ts

```diff
@@ -9,39 +9,40 @@ import { Hex, size } from "viem";
 import { isTableRegistrationLog } from "../isTableRegistrationLog";
 import { logToTable } from "../logToTable";
 import { hexKeyTupleToEntity } from "./hexKeyTupleToEntity";
-import { ConfigToRecsComponents } from "./common";
 import { StorageAdapter, StorageAdapterBlock } from "../common";
-import { configToRecsComponents } from "./configToRecsComponents";
 import { singletonEntity } from "./singletonEntity";
 import storeConfig from "@latticexyz/store/mud.config";
 import worldConfig from "@latticexyz/world/mud.config";
+import { TablesToComponents, tablesToComponents } from "./tablesToComponents";
 
-export type RecsStorageOptions<TConfig extends StoreConfig = StoreConfig> = {
+const storeTables = resolveConfig(storeConfig).tables;
+const worldTables = resolveConfig(worldConfig).tables;
+
+export type RecsStorageOptions<tables extends Record<string, Table>> = {
   world: RecsWorld;
-  // TODO: make config optional?
-  config: TConfig;
+  tables: tables;
   shouldSkipUpdateStream?: () => boolean;
 };
 
-export type RecsStorageAdapter<TConfig extends StoreConfig = StoreConfig> = {
+export type RecsStorageAdapter<tables extends Record<string, Table>> = {
```

Comment:
I'm still not used to `tables` instead of `TTables` 😭 Don't you find it confusing not to see on first sight whether you have to use `typeof tables` vs just `tables`? I'm just so used to custom types being uppercase and only primitive types being lowercase